goss_config.go

```diff
@@ -49,64 +49,15 @@ func NewGossConfig() *GossConfig {
 // Merge consumes all the resources in g2 into c, duplicate resources
 // will be overwritten with the ones in g2
 func (c *GossConfig) Merge(g2 GossConfig) {
-	for k, v := range g2.Files {
-		c.Files[k] = v
-	}
-
-	for k, v := range g2.Packages {
-		c.Packages[k] = v
-	}
-
-	for k, v := range g2.Addrs {
-		c.Addrs[k] = v
-	}
-
-	for k, v := range g2.Ports {
-		c.Ports[k] = v
-	}
-
-	for k, v := range g2.Services {
-		c.Services[k] = v
-	}
-
-	for k, v := range g2.Users {
-		c.Users[k] = v
-	}
-
-	for k, v := range g2.Groups {
-		c.Groups[k] = v
-	}
-
-	for k, v := range g2.Commands {
-		c.Commands[k] = v
-	}
+	v1 := reflect.Indirect(reflect.ValueOf(c))
+	v2 := reflect.ValueOf(g2)
+	for i := 0; i < v1.NumField(); i++ {
+		f1 := v1.Field(i)
+		f2 := v2.Field(i)
+		for _, k := range f2.MapKeys() {
+			f1.SetMapIndex(k, f2.MapIndex(k))
```

Comment:
@aelsabbahy this looks almost taken from the standard library! So I don't think you are working in the wrong direction. 

I, however, would add a type check between `f1` and `f2` to ensure they are the type.